src/Regex/Util.elm

```diff
@@ -20,19 +18,78 @@ firstSubmatch regex str =
 
 newline : String
 newline =
-    "\x0D?\n"
+    "\u{000D}?\n"
 
 
 replaceAllWith : Regex -> String -> String -> String
-replaceAllWith regex with =
-    Regex.replace Regex.All regex (always with)
+replaceAllWith regex with string =
+    Regex.replace regex (always with) string
 
 
 {-| Replace all lines matching a regex. Unlike `replaceAllWith`, `^` and `$` will match the beginning and end of line instead of the whole input.
 -}
 replaceLinesWith : Regex -> String -> String -> String
 replaceLinesWith regex with =
     String.lines
-        >> replaceIf (Regex.contains regex) with
+        >> List.map (replaceIf (Regex.contains regex) with)
         >> String.join "\n"
         >> unindent
+
+
+replaceIf : (String -> Bool) -> String -> String -> String
+replaceIf fn with string =
+    if fn string then
+        with
+
+    else
+        string
+
+
+{-| Remove the shortest sequence of leading spaces or tabs on each line
+of the string, so that at least one of the lines will not have any
+leading spaces nor tabs and the rest of the lines will have the same
+amount of indentation removed.
+unindent " Hello\\n World " == "Hello\\n World"
+unindent "\\t\\tHello\\n\\t\\t\\t\\tWorld" == "Hello\\n\\t\\tWorld"
+
+Borrowed from <https://github.com/elm-community/string-extra/blob/1.5.0/src/String/Extra.elm>.
+
+This might be better in `String.Util`, but there's a circular dependency, so for
+now it's here.
+
+-}
+unindent : String -> String
+unindent multilineSting =
```

Comment:
`string-extra` has a 0.19 release as of yesterday, so this could be removed. Its tests are not passing because one of its test dependencies hasn't been upgraded, but it's probably safe to use.

Reply:
I think removing dependencies is actually nice. I don't worry about this extra function here.